File: mvp/app/src/main/java/br/com/example/hamburgos/request/PromotionPresenter.java
package br.com.example.hamburgos.request;

import android.support.annotation.NonNull;

import java.util.List;

import br.com.example.hamburgos.model.Promotion;
import br.com.example.hamburgos.ui.PromotionActivity;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class PromotionPresenter {

    private Repository repository;

    private PromotionActivity activity;

    public PromotionPresenter(PromotionActivity activity) {
        this.activity = activity;
        this.repository = new Repository(activity);
    }


    public void getPromotions() {
        repository.listPromotions(new Callback<List<Promotion>>() {
            @Override
            public void onResponse(@NonNull Call<List<Promotion>> call, @NonNull Response<List<Promotion>> response) {
                if ( activity != null) {
                    List<Promotion> promotions = response.body();
                    activity.setContent(promotions);
                }
            }

            @Override
            public void onFailure(@NonNull Call<List<Promotion>> call, Throwable t) {
                activity.setError("Error occur!");
            }
        });
    }

}

File: app/src/main/java/br/com/example/hamburgos/request/PromotionPresenter.kt
package br.com.example.hamburgos.request

import br.com.example.hamburgos.model.Promotion
import br.com.example.hamburgos.ui.PromotionActivity
import br.com.example.hamburgos.util.Constants
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class PromotionPresenter(private val activity: PromotionActivity?) {

    private val repository: Repository = Repository(activity)

    fun getPromotions() {
        repository.listPromotions(object : Callback<List<Promotion>> {
            override fun onResponse(call: Call<List<Promotion>>, response: Response<List<Promotion>>) {
                if (activity != null) {
                    val promotions = response.body()
                    activity.setContent(promotions)
                }
            }

            override fun onFailure(call: Call<List<Promotion>>, t: Throwable) {
                activity!!.setError(Constants.ERROR)
            }
        })
    }

}